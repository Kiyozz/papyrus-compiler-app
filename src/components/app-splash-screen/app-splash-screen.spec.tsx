import { shallow } from 'enzyme'
import React from 'react'
import { CSSTransition } from 'react-transition-group'
import AppSplashScreen from './app-splash-screen'

describe('<AppSplashScreen>', () => {
  it('renders correctly', () => {
    const component = (
      <AppSplashScreen initialized={true} />
    )

    expect(component).toBeTruthy()
  })

  it('matches snapshot when initialized', () => {
    const component = shallow((
      <AppSplashScreen initialized={true} />
    ))

    expect(component).toMatchSnapshot()

    const foundCSSTransition = component.find(CSSTransition)

    expect(foundCSSTransition.prop('in')).toBe(false)
    expect(foundCSSTransition.prop('mountOnEnter')).toBe(true)
    expect(foundCSSTransition.prop('unmountOnExit')).toBe(true)
  })

  it('matches snapshot when not initialized', () => {
    const component = shallow((
      <AppSplashScreen initialized={false} />
    ))

    expect(component).toMatchSnapshot()

    const foundCSSTransition = component.find(CSSTransition)

    expect(foundCSSTransition.prop('in')).toBe(true)
    expect(foundCSSTransition.prop('mountOnEnter')).toBe(true)
    expect(foundCSSTransition.prop('unmountOnExit')).toBe(true)
  })
})
