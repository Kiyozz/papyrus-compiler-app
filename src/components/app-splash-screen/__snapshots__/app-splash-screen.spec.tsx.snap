// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppSplashScreen> matches snapshot when initialized 1`] = `
<CSSTransition
  classNames="app-fade"
  in={false}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="app-splash-screen"
  >
    <img
      alt="Application logo"
      src="papyrus-compiler-app-@2x.png"
    />
  </div>
</CSSTransition>
`;

exports[`<AppSplashScreen> matches snapshot when not initialized 1`] = `
<CSSTransition
  classNames="app-fade"
  in={true}
  mountOnEnter={true}
  timeout={300}
  unmountOnExit={true}
>
  <div
    className="app-splash-screen"
  >
    <img
      alt="Application logo"
      src="papyrus-compiler-app-@2x.png"
    />
  </div>
</CSSTransition>
`;
